packages/remix-dev/cli/create.ts

```diff
@@ -284,10 +285,40 @@ async function downloadAndExtractTarball(
     filePath?: string | null;
   }
 ): Promise<void> {
-  let response = await fetch(
-    url,
-    token ? { headers: { Authorization: `token ${token}` } } : {}
-  );
+  let resourceUrl = url;
+  let headers: Record<string, string> = {};
+  if (isGithubReleaseAssetUrl(url)) {
+    // We can download the asset via the github api, but first we need to look up the
+    // asset id
+    let info = getGithubReleaseAssetInfo(url);
+    headers = {
+      Authorization: `token ${token}`,
+      Accept: "application/vnd.github.v3+json",
+    };
+    let response = await fetch(
+      `https://api.github.com/repos/${info.owner}/${info.name}/releases/tags/${info.tag}`,
+      { headers }
+    );
+    if (response.status !== 200) {
+      throw Error(
+        "🚨 There was a problem fetching the file from GitHub. The request " +
+          `responded with a ${response.status} status. Please try again later.`
+      );
+    }
+    let body = await response.json();
+    let assetId: number | undefined = body?.assets?.find(
+      (a: any) => a?.browser_download_url === url
+    )?.id;
+    if (!assetId) {
+      throw Error(
+        "🚨 There was a problem fetching the file from GitHub. No asset was " +
+          "found at that url. Please try again later."
+      );
+    }
+    resourceUrl = `https://api.github.com/repos/${info.owner}/${info.name}/releases/assets/${assetId}`;
+    headers.Accept = "application/octet-stream";
+  }
+  let response = await fetch(resourceUrl, { headers });
```

Comment:
This still doesn't work for me in the private repo case (I'm just pointing to `machour/remix-starter` which doesn't have releases, so it isn't affected by the previous if branch).

In my case, headers are still an empty object. It works when I add this before this fetch:

```js
 headers = {
      Authorization: `token ${token}`,
      Accept: "application/vnd.github.v3+json",
 };
```

Could you provision headers before the `isGithubReleaseAssetUrl()` test?

Reply:
Fixed. Looks like there was a bug in my github-mocks that was hiding this issue. Thanks for pointing it out :)